mantle/kola/tests/misc/boot-mirror.go

```diff
@@ -203,6 +204,14 @@ func detachPrimaryBlockDevice(c cluster.TestCluster, m platform.Machine) {
 		if err := m.(platform.QEMUMachine).RemovePrimaryBlockDevice(); err != nil {
 			c.Fatalf("failed to delete the first boot disk: %v", err)
 		}
+		// Check for the boot ID before initiating a reboot
+		for {
```

Comment:
Rather than indefinitely retrying here, we should use something like `util.Retry` to limit it to something reasonable.